From b24fa3c8b1401a0a15f466843f6c055491644036 Mon Sep 17 00:00:00 2001
From: Martin Lund <malu@gomspace.com>
Date: Fri, 22 Jun 2018 08:00:25 +0000
Subject: [PATCH] zynqmp-zcu102: Enable has-io-mode and 50 MHz max spi
 frequency

Forces non-dma mode. Required tp run UBIFS successfully.
---
 arch/arm64/boot/dts/xilinx/zynqmp-zcu102-revA.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-zcu102-revA.dts b/arch/arm64/boot/dts/xilinx/zynqmp-zcu102-revA.dts
index dbaac91de06d..a29a4b6808da 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-zcu102-revA.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-zcu102-revA.dts
@@ -835,6 +835,7 @@ drivers/hwmon/pmbus/Makefile:11:obj-$(CONFIG_SENSORS_MAX20751)  += max20751.o
 &qspi {
 	status = "okay";
 	is-dual = <1>;
+	has-io-mode;
 	flash@0 {
 		compatible = "m25p80"; /* 32MB */
 		#address-cells = <1>;
@@ -842,7 +843,7 @@ drivers/hwmon/pmbus/Makefile:11:obj-$(CONFIG_SENSORS_MAX20751)  += max20751.o
 		reg = <0x0>;
 		spi-tx-bus-width = <1>;
 		spi-rx-bus-width = <4>; /* FIXME also DUAL configuration possible */
-		spi-max-frequency = <108000000>; /* Based on DC1 spec */
+		spi-max-frequency = <50000000>; /* 50 Mhz - safe frequency */
 		partition@qspi-fsbl-uboot { /* for testing purpose */
 			label = "qspi-fsbl-uboot";
 			reg = <0x0 0x100000>;
-- 
2.17.1

